src/components/multi_select_state.js

```diff
@@ -22,21 +22,23 @@ const withMultiSelectState = WrappedComponent =>
       this.handleChange = this.handleChange.bind(this);
       this.getList = this.getList.bind(this);
       this.onKeyUp = this.onKeyUp.bind(this);
+      this.filterDestinationItems = this.filterDestinationItems.bind(this);
 
       const { items, selectedItems } = props;
       this.state = {
         selectedItems,
         items,
-        filteredItems: items
+        filteredItems: items,
+        filteredDestinationItems: []
       };
     }
 
     componentWillReceiveProps(nextProps) {
-      if (this.props.selectedItems !== nextProps.selectedItems) {
+      if (this.props.selectedItems.length !== nextProps.selectedItems.length) {
```

Comment:
this is not right comparison - arrays sizes can match but the content not.
Why do you need this comparison?